File: doc-examples/jdbc-example-records-java/src/main/java/example/CartRepository.java
package example;

import org.jspecify.annotations.NonNull;
import io.micronaut.data.annotation.Join;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;

import java.util.Optional;

@JdbcRepository(dialect = Dialect.ORACLE)
public interface CartRepository extends CrudRepository<Cart, Long> {

    @Join("items")
    @Override
    Optional<Cart> findById(@NonNull Long id);
}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/CartRepository.kt
package example

import io.micronaut.data.annotation.Join
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.kotlin.KotlinCrudRepository
import java.util.*

@JdbcRepository(dialect = Dialect.H2)
interface CartRepository : KotlinCrudRepository<Cart, Long> {
    @Join("items")
    override fun findById(id: Long): Cart?
}
